sympy/tensor/array/expressions/__init__.py

```diff
@@ -0,0 +1,108 @@
+r"""
+Array expressions are expressions representing N-dimensional arrays, without evaluating them.
+They are *lazy* as they don't evaluate the operations on the arrays until required and
+can define operations on unknown array (in a certain way reminiscent of graphs of
+operations in TensorFlow).
+
+Every N-dimensional array operator has a corresponding lazy object.
+
+Table of correspondences:
+
+=============================== =============================
+         Array operator           Array expression operator
+=============================== =============================
+        tensorproduct                 ArrayTensorProduct
+        tensorcontraction             ArrayContraction
+        tensordiagonal                ArrayDiagonal
+        permutedims                   PermuteDims
+        derive_by_array               array_derive
+=============================== =============================
+
+Examples
+========
+
+>>> from sympy import MatrixSymbol
+>>> from sympy.tensor.array.expressions import ArrayTensorProduct, ArrayContraction
+>>> M = MatrixSymbol("M", 3, 3)
```

Comment:
I would prefer examples for `Array` and `ArraySymbol`, since this is the documentation about arrays. Then competability with matrices can be introduced after it is clear how `Array`s work.